make/photon/prepare/utils/misc.py

```diff
@@ -84,6 +84,22 @@ def prepare_config_dir(root, *name):
         os.makedirs(absolute_path)
     return absolute_path
 
+def prepare_dir(root: str, *args, **kwargs) -> str:
+    gid, uid = kwargs.get('gid'), kwargs.get('uid')
+    absolute_path = Path(os.path.join(root, *args))
+    if absolute_path.is_file():
+        raise Exception('Path exists and the type is regular file')
+
+    absolute_path.mkdir(0o755, parents=True, exist_ok=True)
+
+    if not(gid is None or uid is None):
```

Comment:
Could you update the code so when only `gid` or `uid` is set, the change can also be applied ?